src/strings/kmp_substring_search.jl

```diff
@@ -0,0 +1,101 @@
+"""
+    GetIndexWithKMP(string::String, sub_string::String, ignore_case::Bool)::Int
+
+Find if a string contain a given sub string with KMP Search, as an explanation might be too lengthy
+As such, a detailed explanation can be found at https://towardsdatascience.com/pattern-search-with-the-knuth-morris-pratt-kmp-algorithm-8562407dba5b
+
+
+Questions and answers
+1. Why KMP instead of naive search?
+- Given n = length of string , m = length of string P
+- Then, the naive search will have time complexity O(mn) to find all occurrences of pattern P in S
+- However, knuth-morris-pratt (KMP) algorithm will have tim complexity of O(m+n) to find all occurrences of pattern P in S
+- Hence, KMP is more efficient than the naive approach
+
+
+Example
+
+```julia
+GetIndexWithKMP("Hello I am Gervin", "Hello I am Gervin", false) # returns 1
+GetIndexWithKMP("ABABDABACDABABCABAB", "ABABCABAB", false) # returns 11
+GetIndexWithKMP("SeEms Like IgNOrE CaSe Work", "seems like ignore case work", true) # returns 1
+```
+
+Contributed By:- [Gervin Fung](https://github.com/GervinFung)
+
+
+Note: This function will also allow ignore case also
```

Comment:
```suggestion
Note: This function will also allow ignoring cases also
```

Reply:
Thank you for pointing out the grammar mistake in the documentation! I had already changed it as well.